lib/puppet/provider/package/zypper.rb

```diff
@@ -123,4 +123,9 @@ def update
     # zypper install can be used for update, too
     self.install
   end
+
+  def uninstall
+    options = ['--no-confirm', :remove, @resource[:name]]
```

Comment:
@abednarik no problem! One last thing. With the options specified like this, Puppet calls zypper with:

```
Debug: Executing: '/usr/bin/zypper --no-confirm remove emacs'
Error: Execution of '/usr/bin/zypper --no-confirm remove emacs' returned 2: Unknown option '--no-confirm'
```
`--no-confirm` looks to be a subcommand of `remove`, so we need to be calling `zypper remove --no-confirm emacs`.
If you move the '--no-confirm' to be the second element of the options array, it'll work correctly:

```
options = [:remove, '--no-confirm', @resource[:name]]
```

You'll also need to change the order in the spec test.